client/dashboard/domains/domain-diagnostics/index.tsx

```diff
@@ -0,0 +1,198 @@
+		const uppercaseRecord = recordType.toUpperCase();
+		const record = emailDnsDiagnostics.records[ recordType ];
+
+		if ( ! record ) {
+			return null;
+		}
+
+		let message: string | null = null;
+
+		if ( record.error_message ) {
+			message = record.error_message;
+		} else if ( record.status === 'incorrect' ) {
+			message = sprintf(
+				/* translators: dnsRecordType is a DNS record type, e.g. SPF, DKIM or DMARC */
+				__( 'Your %(dnsRecordType)s record is incorrect. The correct record should be:' ),
+				{
+					dnsRecordType: uppercaseRecord,
+				}
+			);
+		} else if ( record.status === 'not_found' ) {
+			message = sprintf(
+				/* translators: dnsRecordType is a DNS record type, e.g. SPF, DKIM or DMARC */
+				__( 'There is no %(dnsRecordType)s record. The correct record should be:' ),
+				{
+					dnsRecordType: uppercaseRecord,
+				}
+			);
+		}
+
+		if ( ! message ) {
+			return null;
+		}
+
+		return (
+			<li key={ recordType } className="domain-diagnostics-notice__record-item">
+				<Text>{ message }</Text>
+				{ record.correct_record && (
+					<pre>
+						<code>{ record.correct_record }</code>
+					</pre>
+				) }
+			</li>
+		);
+	};
+
+	const renderDiagnostics = () => {
+		const emailDnsDiagnostics = domainDiagnostics?.email_dns_records;
+
+		if (
+			! emailDnsDiagnostics ||
+			emailDnsDiagnostics.code === 'domain_not_mapped_to_atomic_site' ||
+			emailDnsDiagnostics.all_essential_email_dns_records_are_correct
+		) {
+			return (
+				<Notice
+					variant="success"
+					title={ __( 'We didn’t find any issues with your domain!' ) }
+				></Notice>
+			);
+		}
+
+		const recordsToCheck = [ 'spf', 'dkim1', 'dkim2', 'dmarc' ];
+
+		return (
+			<VStack>
+				<Notice variant="error" title={ __( 'Missing or invalid DNS records' ) }>
+					<Text>
+						{ createInterpolateElement(
+							__(
+								'If you use this domain name to send email from your WordPress.com website, the following email records are required. <LearnMoreLink />'
+							),
+							{
+								LearnMoreLink: <InlineSupportLink supportContext="domain-email-authentication" />,
+							}
+						) }
+					</Text>
+				</Notice>
+
+				<VStack as="ul" spacing={ 2 } className="domain-diagnostics-notice__record-list">
+					{ recordsToCheck.map( renderDiagnosticForRecord ) }
+				</VStack>
+
+				{ ! emailDnsDiagnostics.is_using_wpcom_name_servers && (
+					<Notice variant="warning" title={ __( 'Missing or invalid DNS records' ) }>
+						<Text>
+							{ __(
+								'To fix these issues, you should go to your domain’s DNS provider and add the records above to your domain’s DNS settings.'
+							) }
+						</Text>
+					</Notice>
+				) }
+
+				<HStack justify="start" spacing={ 2 }>
+					{ emailDnsDiagnostics.should_offer_automatic_fixes && (
+						<Button
+							variant="primary"
+							onClick={ () =>
+								fixDnsIssues( undefined, {
+									onSuccess: () => {
+										refetchDomainDiagnostics();
```

Comment:
We're already invalidating the query inside `domainDnsEmailMutation()`; is it still necessary?